inst/csv/replacementPatterns.csv

```diff
@@ -18,20 +18,18 @@ oracle,"DATEADD(m,@months,@date)","ADD_MONTHS(@date, @months)"
 oracle,"DATEADD(year,@years,@date)","ADD_MONTHS(@date, 12 * @years)"
 oracle,"DATEADD(yyyy,@years,@date)","ADD_MONTHS(@date, 12 * @years)"
 oracle,"DATEADD(yy,@years,@date)","ADD_MONTHS(@date, 12 * @years)"
-oracle,"DATEDIFF(day,@start, @end)",(CAST(@end AS DATE) - CAST(@start AS DATE))
-oracle,"DATEDIFF(dd,@start, @end)",(CAST(@end AS DATE) - CAST(@start AS DATE))
-oracle,"DATEDIFF(d,@start, @end)",(CAST(@end AS DATE) - CAST(@start AS DATE))
+oracle,"DATEDIFF(day,@start, @end)",(@end - @start)
+oracle,"DATEDIFF(dd,@start, @end)",(@end - @start)
+oracle,"DATEDIFF(d,@start, @end)",(@end - @start)
 oracle,"CONVERT(VARCHAR,@date,112)","TO_CHAR(@date, 'YYYYMMDD')"
 oracle,GETDATE(),SYSDATE
 oracle,+ '@a',|| '@a'
 oracle,'@a' +,'@a' ||
 oracle,CAST(@a AS varchar(@b)) +,CAST(@a AS varchar(@b)) ||
 oracle,+ CAST(@a AS varchar(@b)),|| CAST(@a AS varchar(@b))
-oracle,CAST(@a AS varchar) +,CAST(@a AS varchar) ||

```

Comment:
Similarly, I'm struggling to see how this relates to the issue. The rules here are specifically aimed at converting the '+' sign to a '||' token.

Reply:
Cast(value as varchar) is not a valid syntax in Oracle, so this rule is invalid.  In oracle, all varchar must be followed by a length. 

In the case where we have a CAST(@a as Varchar) it makes sense to do the translated form as TO_CHAR(@a).  Note, you do not want to change the general cast(@a as varchar(#)) into TO_CHAR() because you loose the varchar length (which is the reason for the original issue and this PR).